app/controllers/people_skills_controller.rb

```diff
@@ -28,7 +27,14 @@ def fetch_entries
     if params.key?(:person_id)
       people_skills.where(person_id: params[:person_id])
     elsif params.key?(:skill_id)
-      people_skills.where(skill_id: params[:skill_id])
+      filter_entries(people_skills)
+    end
+  end
+
+  def filter_entries(people_skills)
+    people_skills = people_skills.where(skill_id: params[:skill_id])
+    if params.key?(:level)
+      people_skills.where('level >= ?', params[:level])
     end
   end
```

Comment:
Overloading methods in ruby is rather rare...
Just pass the params[:level] to the PeopleSkillsFilter on line 26 and handle the filter logic in there
Same as with the rated param. Create a new private method `filter_by_level` and filter there